src/kpi/statistics/couplingOfComponents/lib/prFilesUtil.ts

```diff
@@ -0,0 +1,86 @@
+import { ObjectId } from 'mongoose';
+import { coupling } from '../model/coupling';
+
+/**
+ * It computes the coupling over all PRs and their changed files
+ * from @param prFiles coupling size of @param couplingSize and
+ * it considers occurences of the couplings of at least @param occs.
+ * @returns an array with all couplings, occurences and related PRs.
+ */
+export function getCoupling(
+  prFiles: { _id: ObjectId; changedFiles: string[] }[],
+  couplingSize: number = 3,
+  occs: number = 3,
+): coupling[] {
+  const combinationMap = new Map<
+    string,
+    { occs: number; pullrequests: string[] }
+  >();
+
+  prFiles.forEach((diff) => {
+    const combinations = binomialSubsets(diff.changedFiles, couplingSize);
+    const pr = diff._id.toString();
+    combinations.forEach((combo) => {
+      const key: string = combo.sort().join(' ');
+      if (combinationMap.has(key)) {
+        increaseCounterAndAddPR(key, pr);
+      } else {
+        addFileCombinationToMap(key, pr);
+      }
+    });
+  });
+
+  deleteOccurences(occs);
+
+  // sorting map entries descending by occurences
+  const res = [...combinationMap.entries()].sort(
+    (a, b) => b[1]['occs'] - a[1]['occs'],
+  );
+
+  return res;
+
+  function increaseCounterAndAddPR(key: string, pr: string) {
+    const newAmount = combinationMap.get(key).occs + 1;
+    const newPRs = [...combinationMap.get(key).pullrequests, ...[pr]];
```

Comment:
you can just combinationMap.get(key).pullrequests.push(pr)
this should update the array stored in the map.
regardless of that, I'm not sure why you create a new string array by using '[pr]' and then unfold the created array with the spread operator ... 
this looks kind of wrong

Reply:
I remember the problem with this one; push() returns a number, I don't know why.
And therefore I found that way handy to concat to arrays, the old one which is stored already in the map with the new one, which is just a string of course, but regarding the push function we want to have it as an array.
I am open for a better way of handling this, but it works.

Reply:
it returns the new length of the array. so u actually update the array reference on which you push. so you don't need to reassign the array in the map. 
just push the objects in there and increase the number of occurrences. 

code example [here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAtgTwLIEMAOMC8MwFMDuMqaAPNAE4CWYA5gDQwDeoArmFLuQFw4twBGnBmnIQeFajQDaAXQC+APgAUASgBQaxMQB0EXFCUAiDtEMNmINh24wATMNE8ph8LjMxj+EIfnq1oSFhgFnJyXHYANRQAGxZcLHhkdG0afSMTKEMVAEJ-cAgQaNxtaJAaJWDQ8Kgo2Nx1SrDImLjtVnZOGABqTABGGABufxCmmpbikQhtNBYIAAt0ubC3BvzC4tLyrWTUg2NcUxV1IA)